### PR TITLE
Let `GradleInspector` filter all dependencies metadata configurations

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt [hunk 1]
```diff
@@ -44,7 +44,10 @@ internal fun Configuration.isRelevant(): Boolean {
     val isDeprecatedConfiguration = GradleVersion.current() >= GradleVersion.version("6.0")
             && this is DeprecatableConfiguration && resolutionAlternatives != null
 
-    return canBeResolved && !isDeprecatedConfiguration
+    // Do not try to resolve dependencies metadata configurations as there often cause failures with Gradle itself.
+    val isDependenciesMetadata = name.endsWith("DependenciesMetadata")
```
Comment (fviernau) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:48]: Does this string always have a prefix or could the full config name also be "DependenciesMetadata" ? (lower first char)

File: plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt [hunk 2]
```diff
@@ -65,18 +65,7 @@ internal class OrtModelBuilder : ToolingModelBuilder {
...
-                // higher, see [1]. Also see [2] for valuable information in this context.
-                //
-                // [1]: https://developer.android.com/studio/past-releases/past-agp-releases/agp-4-0-0-release-notes#dependency-metadata
-                // [2]: https://gist.github.com/h0tk3y/41c73d1f822378f52f1e6cce8dcf56aa#orgjetbrainskotlinplatformtype
-                filterNot { it.name.endsWith("DependenciesMetadata") }
-            } else {
-                this
-            }
-        }
+        val resolvableConfigurations = project.configurations.filter { it.isRelevant() }
```
Comment (fviernau) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt:68]: out of scope: Maybe reame resolvableConfigurations to relevantConfigurations

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt [hunk 3]
```diff
@@ -44,7 +44,10 @@ internal fun Configuration.isRelevant(): Boolean {
...
+
```
Comment (fviernau) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:49]: could we include a reference in the code comment to what this "*DependenciesMetadata" is about, and maybe also why it is ok to filter it out?

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt (hunk 1 above)
Comment (sschuberth) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:48]: There is always a prefix.

**Review by fviernau — APPROVED** [review]

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt (hunk 3 above)
Comment (sschuberth) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:49]: It's actually quite hard to find information about this in other places than those already given in the commit message (of the original commit that introduced the check for Android). But I've added those links here, too, for completeness.

File: plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt (hunk 2 above)
Comment (sschuberth) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt:68]: Done in a new commit.

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt (hunk 3 above)
Comment (fviernau) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:49]: Thanks!

**Pull request merged**